proposals/threads/Overview.md

```diff
@@ -362,6 +390,81 @@ Assign to `M.[[BufferObject]]` a new [`SharedArrayBuffer`][] whose
 
 Return `ret` as a Number value.
 
+### `WebAssembly.Global` Objects
+
+A `WebAssembly.Global` object contains a single `global` value which can be
+simultaneously referenced by multiple `Instance` objects. Each `Global` object
+has two internal slots:
+
+* \[\[Global\]\]: a [`value`][]
+
+#### `WebAssembly.Global` Constructor
+
+The `WebAssembly.Global` constructor has the signature:
+
+```
+new Global(globalDescriptor)
+```
+
+If the NewTarget is `undefined`, a [`TypeError`][] exception is thrown (i.e.,
+this constructor cannot be called as a function without `new`).
+
+If `Type(globalDescriptor)` is not Object, a [`TypeError`][] is thrown.
+
+Let `typeName` be [`ToString`][]([`Get`][](`globalDescriptor`, `"type"`).
+
+If `typeName` is not one of `"i32"`, `"f32"`, or `"f64"`, throw a [`TypeError`][].
+
+Let `type` be a [`value type`][]:
+
+* If `typeName` is `"i32"`, let `type` be `i32`.
+* If `typeName` is `"f32"`, let `type` be `f32`.
+* If `typeName` is `"f64"`, let `type` be `f64`.
+
+Let `value` be [`ToWebAssemblyValue`][]([`Get`][](`globalDescriptor`,
+`"value"`) coerced to `type`.
+
+Return a new `WebAssembly.Global` instance with \[\[Global\]\] set to `value`.
+
+#### `WebAssembly.Global.prototype [ @@toStringTag ]` Property
+
+TODO
+
+#### `WebAssembly.Global.prototype [ @@toPrimitive ]` Property
+
+TODO
+
+#### `WebAssembly.Global.prototype.value` Property
+
+This property has the attributes { [[Writable]]: `true`, [[Enumerable]]:
+`true`, [[Configurable]]: `false` }.
+
+TODO
+
+### `WebAssembly.Instance` Constructor
+
+For each [`import`][] `i` in `module.imports`:
+
+1. ...
+1. ...
+1. ...
+1. ...
+1. If `i` is a global import:
+   1. If the `global_type` of `i` is `i64`, throw a `WebAssembly.LinkError`. TODO: don't throw?
+   1. If `Type(v)` is a Number, append [`ToWebAssemblyValue`][]`(v)` to `imports`.
+   1. If `Type(v)` is `WebAssembly.Global`, append `v.[[Global]]` to `imports`.
```

Comment:
Hm, just taking the value would copy the state instead of sharing it. The [[Global]] slot will probably need to refer to some notion of abstract Global provided by the core spec to express the desired semantics (i.e., what's called a "global instance" in the spec draft). And if a plain number is supplied, a new such thing has to be created.